at/delete_jobs.cgi

```diff
@@ -1,16 +1,19 @@
 #!/usr/local/bin/perl
 # Delete multiple at jobs
+use strict;
+use warnings;
+our(%access, %text, %in);
 
 require './at-lib.pl';
 &ReadParse();
 &error_setup($text{'deletes_err'});
-@d = split(/\0/, $in{'d'});
+my @d = split(/\0/, $in{'d'});
 @d || &error($text{'deletes_enone'});
 
 # Delete each one
-@jobs = &list_atjobs();
-foreach $d (@d) {
-	($job) = grep { $_->{'id'} eq $d } @jobs;
+my @jobs = &list_atjobs();
+foreach my $d (@d) {
+	my ($job) = grep { $_->{'id'} eq $d } @jobs;
```

Comment:
That's one of the weird Perl quirks that probably shouldn't be used much, as it is so subtle and not particularly intuitive. Especially when there are so many places where list context is used for other purposes...like `my ($one, $two, @therest) = @_`

Ilia, in short, the parentheses force list context (one or more items) on the receiving variable ($job), which would ordinarily have a scalar (one item) context. Perl functions that operate on lists behave differently when the receiver is scalar vs. list; in most cases, as Jamie notes, it gives the number of items in the list. In other cases, it'll give the first or the last item. In some cases this isn't what you want. So, you can tell Perl, "I want the list", which will in this case actually be the whole item it found.

Webmin has some really hairy nested data structures (from a time before Perl had good class support), so it has a lot of interesting uses of context like this. It can be tricky to figure out what's going on. It twists up my brain all the time. (Jamie has a bigger brain than most folks, and can hold all this stuff in his head in a way that I rarely can.)